src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -1,138 +1,433 @@
  * of `Checkbox` components from which a user can make one or multiple
- * selections. Two variants of the MultiSelect component are offered, each with
- * slightly different functionality and requirements.  Because of these
- * differences, the two variants are broken out in separate stories below.
+ * selections. The MultiSelect allows for an optional set of child checkboxes to be passed,
+ * which makes the “parent” checkbox function as a check/uncheck all toggle.
+ * If all of the children checkboxes are checked, the parent isChecked prop will be true.
+ * If only some of the child checkboxes are checked, the parent checkbox will have a isIndeterminate state,
+ * that can be checked or unchecked.
  */
+
 export const MultiSelect = chakra(
   forwardRef<HTMLDivElement, React.PropsWithChildren<MultiSelectProps>>(
-    (props, ref?) => {
+    (props, _ref?) => {
       const {
+        helperText,
         id,
         isBlockElement = false,
         isDefaultOpen = false,
+        isSearchable = false,
+        defaultItemsVisible = 5,
         items,
-        labelText,
-        onApply,
+        listOverflow = "scroll",
+        buttonText,
         onChange,
         onClear,
         onMixedStateChange,
         selectedItems,
-        type,
-        width = "default",
+        width = "full",
         ...rest
       } = props;
 
-      const commonProps = {
-        id,
-        isBlockElement,
-        isDefaultOpen,
-        items,
-        labelText,
-        onClear,
-        selectedItems,
-        type,
+      // Control the open or closed state of the MultiSelect.
+      const [isOpen] = useState(isDefaultOpen);
+      const [itemsList, setItemsList] = useState(items);
+      const [viewAllLabel, setViewAllLabel] = useState("View all");
+      const [listHeight, setListHeight] = useState("215px");
+      const [listItemsCount, setListItemsCount] = useState(defaultItemsVisible);
+
+      // Separate effect for handling listOverflow "scroll"
+      React.useEffect(() => {
+        if (listOverflow === "scroll") {
+          setListHeight("215px");
+          if (isSearchable || items.some((item) => !!item.children)) {
+            setListHeight("275px");
+          }
+          setItemsList(items);
+        }
+      }, [listOverflow, items]);
+
+      // Separate effect for handling listOverflow "expand"
+      React.useEffect(() => {
+        if (listOverflow === "expand") {
+          setListHeight("");
+          if (listItemsCount === defaultItemsVisible) {
+            displayDefaultItems();
+          } else {
+            setItemsList(items);
+            setViewAllLabel("View less");
+          }
+        }
+      }, [listOverflow, listItemsCount, items, defaultItemsVisible]);
+
+      // Sets the selected items count on the menu button.
+      let getSelectedItemsCount: any;
+      if (selectedItems[id]?.items.length > 0) {
+        getSelectedItemsCount = !!(selectedItems[id]?.items.length || 0);
+      }
```

Comment:
This seems unnecessary. If the length is 0, then it is false. Can you try `const getSelectedItemsCount = !!(selectedItems[id]?.items.length || 0);` to remove the "if" statement?